components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4147,6 +4153,62 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String
         return authenticatedUser;
     }
 
+    /**
+     * Creates an instance of AuthenticatedUser{@link AuthenticatedUser} for the given parameters.
+     *
+     * @param username        username of the user
+     * @param userStoreDomain user store domain
+     * @param tenantDomain    tenent domain
+     * @param idpName         idp name
+     * @param accessingOrganization The organization where the user is authorized to access.
+     * @param appTenantID The tenant ID of the application where user get authenticated.
+     * @return an instance of AuthenticatedUser{@link AuthenticatedUser}
+     */
+    public static AuthenticatedUser createAuthenticatedUser(String username, String userStoreDomain, String
+            tenantDomain, String idpName, String accessingOrganization, int appTenantID)
+            throws IdentityOAuth2Exception {
+
+        AuthenticatedUser authenticatedUser = createAuthenticatedUser(username, userStoreDomain, tenantDomain, idpName);
+        /* For organization bound access tokens, the authenticated user should be populated considering
+                    below factors. */
+        if (!OAuthConstants.AuthorizedOrganization.NONE.equals(accessingOrganization)) {
+            authenticatedUser.setAccessingOrganization(accessingOrganization);
+            String userResidentOrg = resolveOrganizationId(tenantDomain);
+            authenticatedUser.setUserResidentOrganization(userResidentOrg);
+            // Set authorized user tenant domain to the tenant domain of the application.
+            authenticatedUser.setTenantDomain(IdentityTenantUtil.getTenantDomain(appTenantID));
+        }
```

Comment:
Seems like we can extract this to a private method since we are using it more than one place.

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2504/commits/13de5fc1f8327d2e23a21a5655193cb857635a69